src/main/java/io/phasetwo/service/resource/OrganizationsResource.java

```diff
@@ -142,4 +161,114 @@ public Response createOrg(@Valid Organization body) {
+    response.entity(keycloakOrgsRepresentation);
+
+    return response.build();
+  }
+
+  @POST
+  @Path("import")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response importOrgs(
+      KeycloakOrgsRepresentation keycloakOrgsRealmRepresentation,
+      @QueryParam("skipMissingMember") Boolean skipMissingMember,
+      @QueryParam("skipMissingIdp") Boolean skipMissingIdp) {
+    log.debugf("Import orgs for %s", realm.getName());
+    if (!(auth.hasCreateOrg() || (auth.hasViewOrgs() && auth.hasManageOrgs()))) {
+      throw new NotAuthorizedException("Insufficient permission to import organization.");
+    }
+
+    var organizations = keycloakOrgsRealmRepresentation.getOrganizations();
+
+    if (CollectionUtil.isEmpty(organizations)) {
+      Response.ResponseBuilder response = Response.noContent();
+      response.type(MediaType.APPLICATION_JSON);
+      return response.build();
+    }
+
+    KeycloakModelUtils.runJobInTransaction(
+        session.getKeycloakSessionFactory(),
+        (session) -> {
+          organizations.forEach(
+              organizationRepresentation ->
+                  createOrganization(
+                      skipMissingMember, skipMissingIdp, session, organizationRepresentation));
+          AdminEventBuilder adminEventClone = adminEvent.clone(session);
+
+          // create import event
+          adminEventClone
+              .resource(ORGANIZATION_IMPORT.name())
+              .operation(OperationType.CREATE)
+              .resourcePath(session.getContext().getUri())
+              .representation(keycloakOrgsRealmRepresentation)
+              .success();
+        });
+
+    Response.ResponseBuilder response = Response.ok();
+    response.type(MediaType.APPLICATION_JSON);
+
+    return response.build();
+  }
+
+  private void createOrganization(
+      Boolean skipMissingMember,
+      Boolean skipMissingIdp,
+      KeycloakSession session,
+      OrganizationRepresentation organizationRepresentation) {
+    try {
+      var org =
+          session
+              .getProvider(OrganizationProvider.class)
+              .createOrganization(
+                  realm, organizationRepresentation.getOrganization().getName(), user, false);
+      KeycloakOrgsImportConverter.setOrganizationAttributes(
+          organizationRepresentation.getOrganization(), org);
+
+      KeycloakOrgsImportConverter.createOrganizationRoles(
+          organizationRepresentation.getRoles(), org);
+
+      KeycloakOrgsImportConverter.createOrganizationIdp(
+          realm, organizationRepresentation.getIdpLink(), org, skipMissingIdp);
+
+      KeycloakOrgsImportConverter.addMembers(
+          session, realm, organizationRepresentation, org, skipMissingMember);
+
+      KeycloakOrgsImportConverter.addInvitations(
+          session, realm, organizationRepresentation, org, skipMissingMember);
+    } catch (ModelDuplicateException e) {
+      throw ErrorResponse.exists(
```

Comment:
Looks like this is where it should be a `409 CONFLICT`. Is that what `ErrorResponse.exists` returns?

Reply:
Exactly. It check if organization duplicates are added and return a 409 error